fastcrypto/src/serde_helpers.rs

```diff
@@ -168,16 +168,24 @@ macro_rules! serialize_deserialize_with_to_from_bytes {
     ($type:ty) => {
         impl ::serde::Serialize for $type {
             fn serialize<S: ::serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-                serializer.serialize_str(&self.encode_base64())
+                match serializer.is_human_readable() {
+                    true => serializer.serialize_str(&self.encode_base64()),
+                    false => serializer.serialize_bytes(&self.as_bytes()),
```

Comment:
pls use SerializationHelper as above
also in the deserializer

Reply:
came across a limitation when using this in sui. since the length as to be known for SerializationHelper, this does not work with enums where each member has variable length. 

```
pub enum Signature {
    Ed25519SuiSignature,
    Secp256k1SuiSignature,
    Secp256r1SuiSignature,
}
```

`serializer.serialize_bytes(&self.as_bytes())` would be one way to work around this. Ideally I want to use this macro to derive ser/de for enum as well.

Reply:
You mean that the macro won't work for this enum, right? but it will work for each of its types.
That's ok, the goal of the macro is to have a consistent serde for the crypto types, ones that have canonical representations.

Reply:
:+1 i will have a separate macro customized for sui then. agreed on having serde in fastcrypto to be the most concise one with known length.